src/main/java/org/prebid/server/auction/StoredRequestProcessor.java

```diff
@@ -107,6 +111,59 @@ public StoredRequestProcessor(ApplicationSettings applicationSettings, Metrics m
         return storedRequestsToBidRequest(ampStoredDataFuture, bidRequest, ampRequestId, Collections.emptyMap());
     }
 
+    /**
+     * Fetches stored request.video and map existing values to imp.id.
+     */
+    Future<VideoStoredDataResult> videoStoredDataResult(List<Imp> imps, List<String> errors, Timeout timeout) {
+        final Map<String, String> storedIdToImpId =
+                mapStoredRequestHolderToStoredRequestId(imps, StoredRequestProcessor::getStoredRequestFromImp)
+                        .entrySet().stream()
+                        .collect(Collectors.toMap(Map.Entry::getValue,
+                                impIdToStoredId -> impIdToStoredId.getKey().getId()));
+
+        return applicationSettings.getStoredData(Collections.emptySet(), storedIdToImpId.keySet(), timeout)
+                .map(storedDataResult -> makeVideoStoredDataResult(storedDataResult, storedIdToImpId, errors));
+    }
+
+    private static VideoStoredDataResult makeVideoStoredDataResult(StoredDataResult storedDataResult,
+                                                                   Map<String, String> storedIdToImpId,
+                                                                   List<String> errors) {
+        final Map<String, String> storedIdToStoredImp = storedDataResult.getStoredIdToImp();
+        final Map<String, Video> impIdToStoredVideo = new HashMap<>();
+
+        for (Map.Entry<String, String> storedIdToImpIdEntry : storedIdToImpId.entrySet()) {
+            final String storedId = storedIdToImpIdEntry.getKey();
+            final String storedImp = storedIdToStoredImp.get(storedId);
+            if (storedImp == null) {
+                errors.add(String.format("No stored Imp for stored id %s", storedId));
+                continue;
+            }
+
+            final String impId = storedIdToImpIdEntry.getValue();
+            final Video video = parseVideoFromImp(storedImp);
+            if (video == null) {
+                errors.add(String.format("No stored video found for Imp with id %s", impId));
+                continue;
+            }
+
+            impIdToStoredVideo.put(impId, video);
+        }
+
+        return VideoStoredDataResult.of(impIdToStoredVideo, errors);
+    }
+
+    private static Video parseVideoFromImp(String storedJson) {
+        if (StringUtils.isNotBlank(storedJson)) {
+            try {
+                final Imp imp = Json.mapper.readValue(storedJson, Imp.class);
+                return imp.getVideo();
+            } catch (JsonProcessingException e) {
+                return null;
```

Comment:
I'm not sure that in case `JsonProcessingException` occurs we should respond with message `"No stored video found"` - I think we should signal that it is invalid if it matters. @rpanchyk what do you think?